rest_framework_json_api/views.py

```diff
@@ -98,12 +101,87 @@ def get_queryset(self, *args, **kwargs):
         return qs
 
 
-class ModelViewSet(AutoPrefetchMixin, PrefetchForIncludesHelperMixin, viewsets.ModelViewSet):
+class RelatedMixin(object):
+    """
+    This mixin handles all related entities, whose Serializers are declared in "related_serializers"
+    """
+    related_serializers = {}
+    related_field_mapping = {}
+
+    def retrieve_related(self, request, *args, **kwargs):
+        serializer_kwargs = {}
+        instance = self.get_related_instance()
+
+        if hasattr(instance, 'all'):
+            instance = instance.all()
+
+        if callable(instance):
+            instance = instance()
+
+        if instance is None:
+            return Response(data=None)
+
+        if isinstance(instance, Iterable):
+            serializer_kwargs['many'] = True
+
+        serializer = self.get_serializer(instance, **serializer_kwargs)
+        return Response(serializer.data)
+
+    def get_serializer_class(self):
+        parent_serializer_class = super(RelatedMixin, self).get_serializer_class()
+
+        if 'related_field' in self.kwargs:
+            field_name = self.kwargs['related_field']
+
+            assert hasattr(parent_serializer_class, 'included_serializers')\
+                or self.related_serializers,\
+                'Either "included_serializers" or ' \
+                '"related_serializers" should be configured'
+
+            # Try get the class from related_serializers
+            class_str = self.related_serializers.get(field_name, None)
+
+            if class_str is None:
+                # Class was not found in related_serializers, look for it in included_serializers
+                class_str = getattr(self, 'included_serializers', {}).get(field_name, None)
+
+                if class_str is None:
+                    raise NotFound
+            return import_class_from_dotted_path(class_str)
+
+        return parent_serializer_class
+
+    def get_related_field_name(self):
+        field_name = self.kwargs['related_field']
+        if field_name in self.related_field_mapping:
+            return self.related_field_mapping[field_name]
+        return field_name
+
+    def get_related_instance(self):
+        parent_obj = self.get_object()
```

Comment:
Nice! Really like it. I think we should now remove the  `related_field_mapping` as not needed in most cases anymore.

We can still leave the method `get_related_field_name` so if someone wants to have its own mapping or whatever logic to get related field name they can overwrite it.